core/src/repair_service.rs

```diff
@@ -518,20 +532,40 @@ impl RepairService {
         slot: Slot,
         slot_meta: &SlotMeta,
         max_repairs: usize,
+        now_timestamp: u64,
+        defer_threshold_ticks: u64,
     ) -> Vec<ShredRepairType> {
         if max_repairs == 0 || slot_meta.is_full() {
             vec![]
         } else if slot_meta.consumed == slot_meta.received {
+            // check delay time of last shred
+            if let Some(reference_tick) = slot_meta
+                .received
+                .checked_sub(1)
+                .and_then(|index| blockstore.get_data_shred(slot, index).ok()?)
+                .and_then(|shred| shred::layout::get_reference_tick(&shred).ok())
+                .map(u64::from)
+            {
+                let ticks_since_first_insert = DEFAULT_TICKS_PER_SECOND
+                    * (now_timestamp.saturating_sub(slot_meta.first_shred_timestamp))
```

Comment:
there seems to be extra parenthesis here.

Reply:
removed